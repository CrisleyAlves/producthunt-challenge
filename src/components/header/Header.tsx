import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";

import { PROFILE_PICTURE } from "@config/constants";
import Avatar from "@components/avatar/Avatar";
import InputDate from "@components/datepicker/DatePIcker";

export default function Header({
  handleOnChangeDate,
  postDate,
  setShowSearchForm,
}) {
  return (
    <div className="header">
      <Avatar
        classNames="avatar--circle"
        avatarUrl={PROFILE_PICTURE}
        altText="profile avatar"
        ariaDescription="profile avatar"
      />

      <InputDate onChangeDate={handleOnChangeDate} currentDate={postDate} />

      <button
        className="search__button"
        aria-describedby="search"
        onClick={setShowSearchForm}
      >
        <FontAwesomeIcon className="icon" icon={faSearch} />
      </button>
    </div>
  );
}
