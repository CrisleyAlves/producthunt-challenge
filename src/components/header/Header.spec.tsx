import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";

import Header from "./Header";

describe("Header", () => {
  const mockHandleOnChangeDate = vi.fn();
  const mockSetShowSearchForm = vi.fn();
  const postDate = new Date("2023-01-01");

  it("renders avatar, date picker, and search button", () => {
    const { container } = render(
      <Header
        handleOnChangeDate={mockHandleOnChangeDate}
        postDate={postDate}
        setShowSearchForm={mockSetShowSearchForm}
      />
    );

    expect(container).toMatchSnapshot();

    expect(screen.getByAltText(/profile avatar/i)).toBeDefined();
    expect(screen.getByRole("textbox")).toBeDefined();
    expect(screen.getByRole("button")).toBeDefined();
  });

  it("calls setShowSearchForm when search button is clicked", () => {
    render(
      <Header
        handleOnChangeDate={mockHandleOnChangeDate}
        postDate={postDate}
        setShowSearchForm={mockSetShowSearchForm}
      />
    );

    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(mockSetShowSearchForm).toHaveBeenCalled();
  });
});
