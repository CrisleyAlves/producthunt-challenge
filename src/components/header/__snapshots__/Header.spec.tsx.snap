// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders avatar, date picker, and search button 1`] = `
<div>
  <div
    class="header"
  >
    <img
      alt="profile avatar"
      aria-description="profile avatar"
      class="avatar avatar--circle"
      src="https://www.gravatar.com/avatar/00000000000000000000000000000000?d=mp&f=y"
    />
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          class="datepicker"
          type="text"
          value="12/31/2022"
        />
      </div>
    </div>
    <button
      aria-describedby="search"
      class="search__button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-magnifying-glass icon"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;
